cmd/earthly/main.go

```diff
@@ -2573,9 +2573,24 @@ func (app *earthlyApp) actionConfig(c *cli.Context) error {
 		return errors.Wrap(err, "read config")
 	}
 
-	outConfig, err := config.UpsertConfig(inConfig, args[0], args[1])
-	if err != nil {
-		return errors.Wrap(err, "upsert config")
+	var outConfig []byte
+
+	switch {
+	case args[1] == "--delete":
```

Comment:
Tbh I think providing the flags as the first argument like `earthly config --delete global.conversion_parallelism` is a bit more natural, but the `--help` tag already worked this way.

Would be easy to support both ways (and maybe deprecate the old way) if we want?

Reply:
I think moving it to https://github.com/earthly/earthly/blob/main/cmd/earthly/main.go#L970-L974 is the right call.

The only reason we support `--help` as a command afterwards is to prevent people from accidentally writting the string `"--help"` as a config value.

As an extra ask, can you extend `case args[1] == "--help":` to include `-h`.

Reply:
> As an extra ask, can you extend case args[1] == "--help": to include -h.

Good call, that's also in the help printout but currently doesn't work on main either.

What do we think about supporting both positions for --help or --delete?

Reply:
I have no strong opinion, supporting both does lead to more code, but perhaps a better user experience. Your call.

Reply:
Supporting both wasn't as easy as it seemed so I figured not worth the effort